### PR TITLE
GLEP 70(?) / Bug 602202: Distribution variables

### PR DESCRIPTION
**Pull request by awilfox**
This PR implements the distribution variables as discussed on [gentoo-dev](https://archives.gentoo.org/gentoo-dev/message/945396b315f11861a6f331bc50416c8f) and [Bug 602202](https://bugs.gentoo.org/show_bug.cgi?id=602202).

I don't necessarily expect this to be merged, but I am putting it here so the wider community may comment.  Thank you.

### CODE REVIEW

File: app-admin/webmin/webmin-1.791.ebuild [hunk 1]
```diff
@@ -285,7 +285,7 @@ pkg_config(){
 	export perl="$( which perl )"
 	export os_type='gentoo-linux'
 	export os_version='*'
-	export real_os_type='Gentoo Linux'
+	export real_os_type="${DISTRO} Linux"
```
Comment (mgorny) [inline app-admin/webmin/webmin-1.791.ebuild]: That looks bad. We certainly shouldn't be forcing 'foo Linux' long form on people. os-release has separate NAME and PRETTY_NAME to account for this.

File: sys-apps/baselayout/baselayout-2.2-r1.ebuild [hunk 2]
```diff
@@ -181,7 +181,7 @@ src_prepare() {
 	echo "LDPATH='${ldpaths#:}'" >> etc/env.d/00basic
 
 	# rc-scripts version for testing of features that *should* be present
-	echo "Gentoo Base System release ${PV}" > etc/gentoo-release
+	echo "${DISTRO} Base System release ${PV}" > etc/gentoo-release
```
Comment (mgorny) [inline sys-apps/baselayout/baselayout-2.2-r1.ebuild]: Not sure about this either. I'd say that if you fork Gentoo, you want to fork baselayout to have proper *-release files for your distro. Maybe we should just assert for Gentoo here.

**Comment by awilfox** [comment]
> I have to say that I still don't like the implementation based on adding yet another redundant location for distribution info. Can't we just make this an eclass reading /etc/os-release?

As mentioned on gentoo-dev, the main issue is for bootstrapping where ``/etc/os-release`` does not yet exist.  It would need to read from ``${ROOT}``, and since this includes packages such as sys-apps/coreutils, it is not guaranteed that baselayout is installed.  Consider also the embedded profile, which has only busybox for all of ``@system``.  It would completely fail if you were attempting to use crossdev or such.

> This would also have the advantage of not forcing distributions to fork the whole profile tree.

Simple ``/etc/portage/make.conf`` setting is enough.  You don't need to fork the whole profile tree.  For an example I have just ran ``emerge -1 coreutils`` after setting ``DISTRO=MyDistro`` in ``/etc/portage/make.conf``:

```
>>> Configuring source in /var/tmp/portage/sys-apps/coreutils-8.26/work/coreutils-8.26 ...
 * econf: updating coreutils-8.26/build-aux/config.sub with /usr/share/gnuconfig/config.sub
 * econf: updating coreutils-8.26/build-aux/config.guess with /usr/share/gnuconfig/config.guess
./configure --prefix=/usr --build=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu --mandir=/usr/share/man --infodir=/usr/share/info --datadir=/usr/share --sysconfdir=/etc --localstatedir=/var/lib --disable-dependency-tracking --disable-silent-rules --libdir=/usr/lib64 --with-packager=MyDistro --with-packager-version=8.26 (p1.1) --with-packager-bug-reports=http://company.internal/bugs --enable-install-program=arch,, --enable-no-install-program=groups,hostname,kill,su,uptime --enable-largefile --disable-libcap --enable-nls --enable-acl --disable-single-binary --enable-xattr --without-gmp
```

File: sys-apps/baselayout/baselayout-2.2-r1.ebuild (hunk 2 above)
Comment (awilfox) [inline sys-apps/baselayout/baselayout-2.2-r1.ebuild]: Understandable, and agreed.  We have [adelie-base](https://code.foxkit.us/adelie/packages/tree/master/sys-apps/adelie-base) and add sys-apps/baselayout to package.provided in Adélie; I will undo this change.

File: app-admin/webmin/webmin-1.791.ebuild (hunk 1 above)
Comment (awilfox) [inline app-admin/webmin/webmin-1.791.ebuild]: I'm not sure the best way to proceed with this; waiting for reply to my comment first.

**Comment by mgorny** [comment]
Yeah, bootstrapping makes things harder. However, can't we just guarantee that baselayout would be installed first? I guess adding a direct DEPEND on baselayout in the eclass could solve that. @gentoo/releng?

**Comment by robbat2** [comment]
@awilfox, @mgorny I thought about both of your sets of discussions here, and suggested some changes to the original proposal, that I think resolves all of your concerns.

Linked from the list:
https://archives.gentoo.org/gentoo-dev/message/1f53fb703a8e559f77fce3146ffbc2ff

**Comment by mgorny** [comment]
What's the status here? Are we continuing the work on this or abandoning the idea?

**Comment by robbat2** [comment]
Needs updating to match the list proposal that seemed to be agreed on (don't know if it needs a GLEP for approval).

**Comment by mgorny** [comment]
Well, it's full of conflicts now and doesn't seem that anything is happening about it. I'm closing it for now, feel free to reopen when it's updated.

**Comment by mgorny** [comment]
(note that if you plan to force push, you need to reopen before doing that)

**Pull request closed without merging**